storage/internal/benchmarks/warmup.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func warmupW1R3(ctx context.Context, opts *benchmarkOptions) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	discardBenchmarkResults(ctx)
+
+	warmupGroup, ctx := errgroup.WithContext(ctx)
+	warmupGroup.SetLimit(opts.numWorkers)
+
+	for deadline := time.Now().Add(opts.warmup); time.Now().Before(deadline); {
+		warmupGroup.Go(func() error {
+			benchmark := &w1r3{opts: opts, bucketName: opts.bucket}
```

Comment:
So, I guess this uses the same level of parallelism as was specified for the main workload? I am wondering if it makes more sense to have a fixed/higher level of parallelism, or to be able to specify this separately.

Reply:
It does; I am open to changing it to a fixed parallelism level - say 16? I think being able to specify this separately may create unnecessary confusion - unless we determine it would be beneficial to have different levels of parallelism for warmups for different workloads, in which case we can always add this in.

The way it is now gives consistency in terms of bucket traffic: the bucket always sees the same level of parallelism; but ultimately for the long term a fixed parallelism makes sense.

Reply:
Yeah I think 16 (or numCPUs) is appropriate. We could make it configurable if/when we refactor all the CLI opts into some kind of config file eventually...

Reply:
Good idea - I changed it to num cpus